### PR TITLE
Add types, move loading logic to loading screen, ran linter

### PR DESCRIPTION
**Pull request by amanji**
Signed-off-by: Akiff Manji <akiff.manji@gmail.com>

### CODE REVIEW

File: src/views/Splash.tsx [hunk 1]
```diff
@@ -1,17 +1,40 @@
...
 
 const container = css`
   flex: 1;
   justify-content: center;
   align-items: center;
 `;
 
-export const Splash = () => {
+export const Splash: React.FC<any> = ({navigation}) => {
+  const [, dispatch] = useContext(Context);
```
Comment (jleach) [inline src/views/Splash.tsx:18]: Clever. Didn't know this trick. 🤓

File: src/views/credential/Credentials.tsx [hunk 2]
```diff
@@ -52,8 +47,8 @@ const extraMarginBottom = css`
   margin-bottom: 24px;
 `;
 
-export const Credentials = ({navigation}) => {
-  const [state, dispatch] = useContext(Context);
+export const Credentials: React.FC<any> = ({navigation}) => {
```
Comment (jleach) [inline src/views/credential/Credentials.tsx:50]: `React.FC` is the return type - no? Don't like it?

**Review by jleach — APPROVED** [review]

File: src/views/credential/Credentials.tsx (hunk 2 above)
Comment (amanji) [reply src/views/credential/Credentials.tsx:50]: I was just being consistent with your components. They all have types.

Comment (jleach) [reply src/views/credential/Credentials.tsx:50]: Oof. 🤦I misread that as a `-`. You were adding not removing.

**Pull request merged**